tensorflow/python/summary/writer/event_file_writer.py

```diff
@@ -108,7 +112,9 @@ def close(self):
 
     Call this method when you do not need the summary writer anymore.
     """
+    self._worker.stop()
     self.flush()
+    self._worker.join()
```

Comment:
Won't this block forever if `self._worker` is blocked in `self._queue.get()` on L151?